### ISSUE TITLE
Identifier_name broken

### ISSUE THREAD

**Issue by paul-lavoine** [realm/SwiftLint#4956]
### New Issue Checklist

- This issue happens since the version 0.51.0
- No issue similar

### Describe the bug

Since the version 0.51.0 **identifier_name** doesn't work as before.
I have seen in the release tag https://github.com/realm/SwiftLint/releases/tag/0.51.0, the string can be interpret has regex in the excluded for the identifier_name (PR: https://github.com/realm/SwiftLint/pull/4655)

But the classic exclusion looks broken, because the following line of code worked in the version 0.50.3

```
identifier_name:
    excluded:
        - _body(configuration:)
```

```
Identifier Name Violation: Function name '_body(configuration:)' should start with a lowercase character (identifier_name)
```

### Environment

* SwiftLint version 0.51.0
* CocoaPods
* Paste your configuration file:

```yml
identifier_name: 
    min_length:
        error: 3
    max_length:
        warning: 60
    excluded: 
        - _body(configuration:)
```

**Comment by SimplyDanny** [comment]
#4655 might indeed have broken your configuration. Could you try escaping the parentheses?

**Comment by paul-lavoine** [comment]
It works, thanks
